crates/router/src/connector/prophetpay/transformers.rs

```diff
@@ -22,122 +28,339 @@ impl<T>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        (_currency_unit, _currency, amount, item): (
+        (currency_unit, currency, amount, item): (
             &types::api::CurrencyUnit,
             types::storage::enums::Currency,
             i64,
             T,
         ),
     ) -> Result<Self, Self::Error> {
+        let amount = utils::get_amount_as_f64(currency_unit, amount, currency)?;
         Ok(Self {
             amount,
             router_data: item,
         })
     }
 }
 
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
-pub struct ProphetpayPaymentsRequest {
-    amount: i64,
-    card: ProphetpayCard,
+pub struct ProphetpayAuthType {
+    pub(super) user_name: Secret<String>,
+    pub(super) password: Secret<String>,
+    pub(super) profile_id: Secret<String>,
 }
 
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
-pub struct ProphetpayCard {
-    name: Secret<String>,
-    number: cards::CardNumber,
-    expiry_month: Secret<String>,
-    expiry_year: Secret<String>,
-    cvc: Secret<String>,
-    complete: bool,
+impl TryFrom<&types::ConnectorAuthType> for ProphetpayAuthType {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        match auth_type {
+            types::ConnectorAuthType::SignatureKey {
+                api_key,
+                key1,
+                api_secret,
+            } => Ok(Self {
+                user_name: api_key.to_owned(),
+                password: key1.to_owned(),
+                profile_id: api_secret.to_owned(),
+            }),
+            _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct ProphetpayTokenRequest {
+    ref_info: String,
+    profile: Secret<String>,
+    entry_method: i8,
+    token_type: i8,
+    card_entry_context: i8,
+}
+
+#[derive(Debug, Clone)]
+pub enum ProphetpayEntryMethod {
+    ManualEntry,
+    CardSwipe,
+}
+
+fn get_entry_method(method: ProphetpayEntryMethod) -> i8 {
+    match method {
+        ProphetpayEntryMethod::ManualEntry => 1,
+        ProphetpayEntryMethod::CardSwipe => 2,
+    }
+}
+
+#[derive(Debug, Clone)]
+#[repr(i8)]
+pub enum ProphetpayTokenType {
+    Normal,
+    SaleTab,
+    TemporarySave,
+}
+
+fn get_token_type(token_type: ProphetpayTokenType) -> i8 {
+    match token_type {
+        ProphetpayTokenType::Normal => 0,
+        ProphetpayTokenType::SaleTab => 1,
+        ProphetpayTokenType::TemporarySave => 2,
+    }
+}
```

Comment:
Instead of method, have an implementation for the enum 
```suggestion
impl ProphetpayTokenType {
    fn get_token_type(&self) -> i8 {
        match self {
        Self::Normal => 0,
        Self::SaleTab => 1,
        Self::TemporarySave => 2,
    }
    }
}
```